addons/mail/static/src/new/thread/message_in_reply_to.js

```diff
@@ -22,10 +22,10 @@ export class MessageInReplyTo extends Component {
             parentMessage.author &&
             (parentMessage.resModel === "mail.channel" || !parentMessage.resModel)
         ) {
-            return `/web/image/res.partner/${parentMessage.author.partner.id}/avatar_128`;
+            return `/web/image/res.partner/${parentMessage.author.id}/avatar_128`;
         }
         if (parentMessage.author && parentMessage.resModel === "mail.channel") {
-            return `/mail/channel/${parentMessage.resId}/partner/${parentMessage.author.partner.id}/avatar_128`;
+            return `/mail/channel/${parentMessage.resId}/partner/${parentMessage.author.id}/avatar_128`;
```

Comment:
if author can be a persona guest both of those URL are wrong in that case

Reply:
Indeed, I didn't change the code, so it was already lacking logic on guest here.